shared/common-adapters/video.desktop.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import * as React from 'react'
+import Measure from 'react-measure'
+import type {Props, State} from './video'
+import * as Styles from '../styles'
+import {getVideoSize} from './video.shared'
+import logger from '../logger'
+
+export default class extends React.PureComponent<Props, State> {
+  state = {
+    containerHeight: 0,
+    containerWidth: 0,
+    loadedVideoSize: false,
+    videoHeight: 0,
+    videoWidth: 0,
+  }
+
+  _mounted = false
+
+  _onContainerResize = ({bounds}) =>
+    this.setState({containerHeight: bounds.height, containerWidth: bounds.width})
+
+  _videoRef: {current: HTMLVideoElement | null} = React.createRef()
+  _videoRefAction = (action: HTMLVideoElement => any) => {
+    if (!this._videoRef.current) {
+      // This can happen in story tests.
+      logger.warn('_videoRef is falsey')
+      return
+    }
+    action(this._videoRef.current)
+  }
+
+  _onVideoLoadedmetadata = ({target}) => {
+    this._mounted &&
+      this.setState({
+        loadedVideoSize: true,
+        // $FlowIssue doesn't know videoHeight
```

Comment:
i think you can just make the type and put it into the params vs doing the flow issue

Reply:
Ha! The syntheticEvent has it typed nicely